OpenAI-DotNet/Common/BaseResponse.cs

```diff
@@ -68,12 +69,86 @@ public abstract class BaseResponse
         [JsonIgnore]
         public string ResetRequests { get; internal set; }
 
+        /// <summary>
+        /// The time until the rate limit (based on requests) resets to its initial state represented as a TimeSpan.
+        /// </summary>
+        [JsonIgnore]
+        public TimeSpan ResetRequestsTimespan { get => ConvertTimestampToTimespan(ResetRequests); }
+
         /// <summary>
         /// The time until the rate limit (based on tokens) resets to its initial state.
         /// </summary>
         [JsonIgnore]
         public string ResetTokens { get; internal set; }
 
+        /// <summary>
+        /// The time until the rate limit (based on tokens) resets to its initial state represented as a TimeSpan.
+        /// </summary>
+        [JsonIgnore]
+        public TimeSpan ResetTokensTimespan { get => ConvertTimestampToTimespan(ResetTokens); }
+
+        /// <summary>
+        /// Takes a timestamp recieved from a OpenAI response header and converts to a TimeSpan
+        /// </summary>
+        /// <param name="timestamp">The timestamp received from an OpenAI header, eg x-ratelimit-reset-tokens</param>
+        /// <returns>A TimeSpan that represents the timestamp provided</returns>
+        /// <exception cref="ArgumentException">Thrown if the provided timestamp is not in the expected format, or if the match is not successful.</exception>
+        private TimeSpan ConvertTimestampToTimespan(string timestamp)
+        {
+            /*
+             * Regex Notes: 
+             *  The gist of this regex is that it is searching for "timestamp segments", eg 1m or 144ms.
+             *  Each segment gets matched into its respective named capture group, from which we further parse out the 
+             *  digits. This allows us to take the string 6m45s99ms and insert the integers into a 
+             *  TimeSpan object for easier use.
+             *  
+             *  Regex Performance Notes, against 100k randomly generated timestamps:
+             *  Average performance: 0.0003ms
+             *  Best case: 0ms
+             *  Worst Case: 15ms
+             *  Total Time: 30ms
+             *  
+             *  Inconsequential compute time
+             */
+            Regex tsRegex = new Regex(@"^(?<hour>\d+h)?(?<mins>\d+m(?!s))?(?<secs>\d+s)?(?<ms>\d+ms)?");
```

Comment:
this can probably be a private readonly field

Reply:
Change committed